chore: create unique entries in history when building

cdrage opened the pull request:
chore: create unique entries in history when building

### What does this PR do?

* Instead of overriding the previous history, create a unique entry
  using name
* Adds 'name' to the interface

### Screenshot / video of UI

<!-- If this PR is changing UI, please include
screenshots or screencasts showing the difference -->


https://github.com/containers/podman-desktop-extension-bootc/assets/6422176/dc2ba8f1-a04b-4219-8edc-0777b45eabf1



### What issues does this PR fix or reference?

<!-- Include any related issues from Podman Desktop
repository (or from another issue tracker). -->

Closes https://github.com/containers/podman-desktop-extension-bootc/issues/201

### How to test this PR?

<!-- Please explain steps to reproduce -->

1. Delete `rm
   ~/.local/share/containers/podman-desktop/extensions-storage/redhat.bootc/history.json`
2. Create multiple builds, they should appear now as multiple builds.

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

cdrage commented:
Note this is a **BREAKING** change.

In the next release we should mention that you must delete: `~/.local/share/containers/podman-desktop/extensions-storage/redhat.bootc/history.json`

This is because we are now adding 'name' to the history.json

benoitf commented:
Can we detect the old file and migrate it? So no breaking change

benoitf commented:
or just delete it if it has invalid / old format

deboer-tim commented:
> or just delete it if it has invalid / old format

Yes, that's what I was thinking as well. Limited users so far, and the only purpose it had was pre-filling things like folder names. Not a huge loss if we just start fresh.

cdrage commented:
Ready for review 🎉 

I have added a function as well that will check to see the old history.json format if there is any missing 'name'.

Instead of deleting the old file, I instead just had it add the 'name' entry to it since it was easy enough to do. Keeping compatibility / history for old versions of this extension.

deboer-tim reviewed (changes requested):
Had a side chat with Charlie. The generated names help us refer to a unique disk image generation, but they don't really help the user (no more meaningful that the other columns) and weren't specified by the user.

I think we should just treat the names like ids (or even rename to id) and remove the Name column so that they aren't exposed. It solves the underlying problem and keeps the table more concise.

cdrage commented:
> Had a side chat with Charlie. The generated names help us refer to a unique disk image generation, but they don't really help the user (no more meaningful that the other columns) and weren't specified by the user.
> 
> I think we should just treat the names like ids (or even rename to id) and remove the Name column so that they aren't exposed. It solves the underlying problem and keeps the table more concise.

Updated to now use `id` for the "unique name" and removed the column from the view. This means we can easily update + build multiple images without an issue.

deboer-tim commented on packages/backend/src/build-disk-image.ts:
```diff
@@ -100,12 +100,13 @@ export async function buildDiskImage(build: BootcBuildInfo, history: History): P
   return extensionApi.window
     .withProgress(
-      { location: extensionApi.ProgressLocation.TASK_WIDGET, title: 'Building disk image ' + build.name },
+      { location: extensionApi.ProgressLocation.TASK_WIDGET, title: 'Building disk image ' + build.id },
```
I think we should keep this one to build.image (or build.image+":"+build.tag). If I launch it several days in a row I'd expect the tasks to still say it's building "image-name", not "image-name 5".

deboer-tim commented on packages/backend/src/build-disk-image.ts:
```diff
@@ -100,12 +100,13 @@ export async function buildDiskImage(build: BootcBuildInfo, history: History): P
+      { location: extensionApi.ProgressLocation.TASK_WIDGET, title: 'Building disk image ' + build.id },
       async progress => {
-        const buildContainerName = build.name.split('/').pop() + bootcImageBuilderContainerName;
+        const buildContainerName = build.id + bootcImageBuilderContainerName;
```
Ok to merge, but ditto here or maybe we find a non-conflicting name later. If there's no current container with a conflict I'd expect the name to be "image-name".

deboer-tim reviewed (approved):
Thanks for the changes. One minor thing I'd like you to fix, but otherwise ready to merge.

cdrage commented:
Fixed! Thanks for the feedback.

Confirmed as well with building that this works / changes are now present in json / etc.
![Screenshot 2024-03-26 at 5 34 02 PM](https://github.com/containers/podman-desktop-extension-bootc/assets/6422176/bc7e8774-50ad-4a84-ae9a-46eb4b46992d)


```
cat /Users/cdrage/.local/share/containers/podman-desktop/extensions-storage/redhat.bootc/history.json
[
  {
    "id": "test-2",
    "image": "quay.io/podai/test",
    "tag": "latest",
    "engineId": "podman.Podman Machine",
    "folder": "/Users/cdrage/bootc123",
    "type": "raw",
    "arch": "amd64",
    "status": "running",
    "timestamp": "2024-03-26T21:33:53.287Z",
    "buildContainerId": "b578d2625e41b8f2319b9ada5e9f14bfd67d6f189b48ae7513ef72c32ee5421a"
  },
  {
    "id": "test",
    "image": "quay.io/podai/test",
    "tag": "latest",
    "engineId": "podman.Podman Machine",
    "folder": "/Users/cdrage/bootc",
    "type": "raw",
    "arch": "amd64",
    "status": "running",
    "timestamp": "2024-03-26T21:33:47.998Z",
    "buildContainerId": "e818b90ca90b80b0f3a04c5c94fcf1f6eec8bb16199291be1070b3fefbda679b"
  }
]%
```
